cmd/zpool/zpool_main.c

```diff
@@ -6909,6 +6920,16 @@ zpool_do_online(int argc, char **argv)
 					(void) printf(gettext("use 'zpool "
 					    "replace' to replace devices "
 					    "that are no longer present\n"));
+				if (flags & ZFS_ONLINE_EXPAND &&
+				    (oldstate == VDEV_STATE_DEGRADED ||
+				    oldstate == VDEV_STATE_HEALTHY)) {
```

Comment:
Why do you need to check the `oldstate` here? Is there a scenario where we wouldn't want to produce this error message/return if the device is not healthy after an expand request (e.g., perhaps if the device is OFFLINE)?

Reply:
Ah, I see from the above discussion (which I didn' t bothered to read before looking at the change :-P ) that you are attempting to narrow the error case here... which seems reasonable. But perhaps change the check to simply:
```
oldstate >= VDEV_STATE_DEGRADED) {
```